### PR TITLE
feat(indexedDB): An implementation of IndexedDB

### PR DESCRIPTION
**Pull request by ttowncompiled**

### CODE REVIEW

File: modules/tactical/src/idb.ts [hunk 1]
```diff
@@ -18,15 +19,23 @@ export interface Idb {
...
-  db: Object = {};
+  db: Object;
+
+  /**
+   * Opens a connection to the Idb database.
+   */
+  open(database: string, stores: string[]) {
+    this.db = {};
+    stores.forEach((store: string) => { this.db[store] = {}; });
+  }
```
Comment (gkalpak) [inline modules/tactical/src/idb.ts]: So, does this mean that only one database is possible.
Trying to access stores in different databases will silently return the same instance (as long as the store names match).

Is this intentional ?

File: modules/tactical/src/idb.ts [hunk 2]
```diff
@@ -55,3 +64,88 @@ export class InMemoryIdb implements Idb {
...
+      }
+    });
+  }
+
+  /**
+   * Retrieve the object associated with a key from the given store, or return
+   * null if the key doesn't exist in that store.
+   */
+  get(store: string, key: string): Observable<Object> {
+    this._db.toPromise().then((idbDatabase) => {console.log(idbDatabase);});
```
Comment (gkalpak) [inline modules/tactical/src/idb.ts]: Should this really be here ?

File: modules/tactical/src/idb.ts (hunk 1 above)
Comment (gkalpak) [inline modules/tactical/src/idb.ts]: This basically means that I can't create two different instances pointing to the same DB.
Is this intentional ?

**Comment by ttowncompiled** [comment]
Blocked on PR: #30

File: modules/tactical/src/idb.ts (hunk 1 above)
Comment (ttowncompiled) [inline modules/tactical/src/idb.ts]: Correct. To have two individual instances pointing to the same DB, you have to instantiate and open two separate objects. Obviously, not for this class specifically, but this class is a testing mock.

**Pull request merged**